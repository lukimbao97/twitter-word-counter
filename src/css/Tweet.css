.tweet-main-container {
  display: flex;
  margin: 10px;
  border-bottom: 2px inset;
}

img {
  border-radius: 50%;
}

.tweet-container {
  max-width: 70%;
  padding: 10px 0 5px 10px;
  font-size: 20px;
}

.name-tweet-wrapper {
  display: flex;
  flex-direction: column;
}

.username-wrapper {
  margin-left: 10px;
  font-weight: bold;
}

.tagname-wrapper {
  margin-left: 10px;
  color: gray;
}
