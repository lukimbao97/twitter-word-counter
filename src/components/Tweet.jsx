import React from "react";
import "../css/Tweet.css";

const Tweet = ({ tagName, userName, tweet, avatar }) => {
  return (
    <div className="tweet-main-container">
      <div>
        <img src={avatar} alt="" />
      </div>
      <div className="name-tweet-wrapper">
        <span className="username-wrapper">{userName}</span>
        <span className="tagname-wrapper">{tagName}</span>
        <div className="tweet-container">{tweet}</div>
      </div>
    </div>
  );
};

export default Tweet;
