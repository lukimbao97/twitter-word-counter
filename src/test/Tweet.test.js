import React from "react";
import renderer from "react-test-renderer";
import Tweet from "../components/Tweet";

// Test matchSnapShot Tweet Component
test("test-changes-from-Tweet-component", () => {
  const component = renderer.create(
    <Tweet
      avatar={"https://source.unsplash.com/random/50x50"}
      tweet={"Testing"}
    />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
