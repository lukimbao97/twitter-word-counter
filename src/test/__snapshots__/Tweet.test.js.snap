// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`test-changes-from-Tweet-component 1`] = `
<div
  className="tweet-main-container"
>
  <div>
    <img
      alt=""
      src="https://source.unsplash.com/random/50x50"
    />
  </div>
  <div
    className="tweet-container"
  >
    Testing
  </div>
</div>
`;
